src/main/java/com/teragrep/pth10/steps/teragrep/TeragrepBloomStep.java

```diff
@@ -162,32 +158,35 @@ private Dataset<Row> estimateSize(Dataset<Row> dataset) {
     }
 
     public Dataset<Row> aggregate(Dataset<Row> dataset) {
-
-        FilterTypes filterTypes = new FilterTypes(this.zeppelinConfig);
-
-        BloomFilterAggregator agg = new BloomFilterAggregator(inputCol, estimateCol, filterTypes.sortedMap());
-
+        final FilterTypes filterTypes = new FilterTypes(this.zeppelinConfig);
+        final SortedMap<Long, Double> map = new TreeMap<>();
+        final List<FilterField> fieldList = filterTypes.fieldList();
+        for (final FilterField field : fieldList) {
+            map.put(field.expected(), field.fpp());
+        }
+        final BloomFilterAggregator agg = new BloomFilterAggregator(inputCol, estimateCol, map);
```

Comment:
I like the idea of having the FilterTypes as a list but it feels weird that then it has to be put into a map instantly anyways like before. Could the BloomFilterAggregator be refactored to work with the list too?

Reply:
Indeed a bit unfortunate result of refactoring but refactored BloomFilterAggregator would require new release of dpf_03. Also would couple dpf_03 to this custom object type a general map might be better for use elsewhere.